splunk_add_on_ucc_framework/tabs/proxy_tab.py

```diff
@@ -0,0 +1,208 @@
+            {"value": "socks4", "label": "socks4"},
+            {"value": "socks5", "label": "socks5"},
+        ],
+    },
+    "defaultValue": "http",
+    "field": "proxy_type",
+}
+
+DEFAULT_HOST = {
+    "type": "text",
+    "label": "Host",
+    "validators": [
+        {
+            "type": "string",
+            "errorMsg": "Max host length is 4096",
+            "minLength": 0,
+            "maxLength": 4096,
+        },
+        {
+            "type": "regex",
+            "errorMsg": "Proxy Host should not have special characters",
+            "pattern": "^[a-zA-Z]\\w*$",
+        },
+    ],
+    "field": "proxy_url",
+}
+
+DEFAULT_PORT = {
+    "type": "text",
+    "label": "Port",
+    "validators": [{"type": "number", "range": [1, 65535], "isInteger": True}],
+    "field": "proxy_port",
+}
+
+DEFAULT_USERNAME = {
+    "type": "text",
+    "label": "Username",
+    "validators": [
+        {
+            "type": "string",
+            "errorMsg": "Max length of username is 50",
+            "minLength": 0,
+            "maxLength": 50,
+        }
+    ],
+    "field": "proxy_username",
+}
+
+DEFAULT_PASSWORD = {
+    "type": "text",
+    "label": "Password",
+    "validators": [
+        {
+            "type": "string",
+            "errorMsg": "Max length of password is 8192",
+            "minLength": 0,
+            "maxLength": 8192,
+        }
+    ],
+    "encrypted": True,
+    "field": "proxy_password",
+}
+
+DEFAULT_DNS_RESOLUTION = {
+    "type": "checkbox",
+    "label": "DNS resolution",
+    "field": "proxy_rdns",
+}
+
+
+class ProxyTab(Tab):
+    @property
+    def tab_type(self) -> Optional[str]:
+        return "proxyTab"
+
+    @classmethod
+    def from_definition(cls, definition: Dict[str, Any]) -> Optional["Tab"]:
+        """
+        This function checks if the definition has type == ProxyTab; if it does, it gets converted;
+        otherwise, it returns None.
+        """
+        if definition.get("type") == "proxyTab":
+            entity = []
+            entity_key_const_dict = {
+                "enable_proxy": DEFAULT_PROXY_ENABLE,
+                "proxy_type": DEFAULT_PROXY_TYPE,
+                "host": DEFAULT_HOST,
+                "port": DEFAULT_PORT,
+                "username": DEFAULT_USERNAME,
+                "password": DEFAULT_PASSWORD,
+                "dns_resolution": DEFAULT_DNS_RESOLUTION,
+            }
+
+            def updating_dictionaries(key_name: str, const: Any) -> None:
+                for key, value in const.items():
+                    nonlocal definition
+                    if key not in definition.get(key_name, {}):
+                        definition[key_name][key] = value
+                entity.append(definition.get(key_name))
+
+            if ("username" not in definition) ^ ("password" not in definition):
+                logger.error("Either of username or password is not mentioned.")
+                sys.exit(1)
```

Comment:
What about raising an exception instead of exiting?

Reply:
I went with exiting, as this is a misconfiguration by the user, and we shouldn't proceed further. However, if we raise an exception, we would have to handle all the way to the first caller and then make a clean exit.